runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -847,6 +847,94 @@ Java_jdk_internal_misc_Unsafe_objectFieldOffset1(JNIEnv *env, jobject receiver,
 	return offset;
 }
 
+/**
+ * Writes modified memory in an address range from cache to main memory.
+ * Uses memory barriers before and after writeback to ensure ordering.
+ * 
+ * On x86, the writeback is done using the CLWB instruction if available
+ * for performance, falling back to CLFLUSHOPT then CLFLUSH otherwise.
+ * 
+ * @param addr address of block to write back to memory
+ * @param len length of block being written
+ */
+void JNICALL
+Java_jdk_internal_misc_Unsafe_writebackMemory(JNIEnv *env, jobject receiver, jlong addr, jlong len)
+{
+#if (defined(J9X86) || defined(J9HAMMER))
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	uintptr_t cacheLineSize = vm->dCacheLineSize;
+
+	if (cacheLineSize > 0) {
+		U_8 *endAddr = (*(U_8 **) &addr) + (*(uintptr_t *) &len) - 1;
+		U_8 *ptr = (U_8 *)((*(uintptr_t *) &addr) & ~(cacheLineSize - 1));
+
+		issueReadWriteBarrier();
```

Comment:
One last thing - I didn't see this was a .cpp file, so you can do the barriers without the overhead of a call.

If you include `AtomicSupport.hpp` then you can call `VM_AtomicSupport::readWriteBarrier` instead of `issueReadWriteBarrier`.